src/containers/BookEdit.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import BookShape from '../shapes/book';
+import { fetchBookEdit, updateBook, persistBook } from '../redux/actions';
+
+class BookEdit extends Component {
+  componentDidMount() {
+    this.props.fetchBook(this.props.match.params.isbn);
+  }
+
+  onChangeInput = (event) => {
+    const newBook = {
+      ...this.props.book,
+      [event.target.name]: event.target.value
+    };
+    this.props.updateBook(newBook);
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.persistBook(this.props.book)
```

Comment:
I always try to abstract away components from the knowledge of underlying actions as much as possible. For e.g. component BookEdit knows it wants to submit the book, but it doesn't have to know that the book is going to be persisted. 

It could have used this.props.onSubmit e.g. its a small detail, which leads to a more reusable code in general, but I am not sure how impotrant that is for this course.

Reply:
This is a very good point. In this project, I'd like to keep the focus on the understandability of how redux works. Changing the name from `persistBook` to `onSubmit` in `mapDispatchToProps` would be more confusing if you ask me. So I'd like to keep it as it is.
If you see another elegant way to include the abstraction, let me know. @kof